Fix deprecated imports of collections.abc classes

jmargeta opened the pull request:
This PR proposes to move all collections.abc imports into the compat module.

In Python 3.3, Collections Abstract Base Classes such as Mapping, Iterable, or MutableSet were moved to the collections.abc module.
For backwards compatibility, are still visible in the collections module through Python 3.7. In the following versions they will be entirely removed.

This should fix Issue https://github.com/marshmallow-code/marshmallow/issues/1027
See also: https://docs.python.org/3.7/library/collections.html

jmargeta committed:
Fix deprecation warnings when importing from collections

Import of abc classes directly from collections module was
deprecated in Python 3.6 and will be completely removed
in Python 3.8.

This commit prevents the deprecation warnings and helps
with future compatibility.

lafrech commented:
I suppose the import in schema.py should be changed as well: https://github.com/marshmallow-code/marshmallow/blob/2.x-line/marshmallow/schema.py#L5.

------------------------------------------------------------

I don't see the need for those statements:

```py
    OrderedDict = OrderedDict
    Mapping = Mapping
    Iterable = Iterable
    MutableSet = MutableSet
```

Readability, I guess.

Anyway, this is not specific to this PR. You conformed to the existing convention.

jmargeta committed:
Import Mapping from compat instead of collections

Adds missing import to avoid deprecated import
of ABC classes when imported directly from the
collections module in Python 3.7.

jmargeta commented:
Thanks for looking into the PR @lafrech . I forgot to stage that file in the first commit. Should be fixed now.

I also find that convention unnecessary but thought that there might have been a reason for that.
Will open another PR with that change in a moment (when/if this one gets merged) and we can discuss it there.

sloria commented on marshmallow/compat.py:
```diff
@@ -45,6 +49,13 @@ def get_func_args(func):
     zip_longest = itertools.zip_longest
     from collections import OrderedDict
     OrderedDict = OrderedDict
+    try:
```
This try-except should not be necessary, since we don't support Python 3.3.

sloria commented on marshmallow/compat.py:
```diff
@@ -45,6 +49,13 @@ def get_func_args(func):
+        from collections import Mapping, Iterable, MutableSet
+    except ImportError:
+        from collections.abc import Mapping, Iterable, MutableSet
+    Mapping = Mapping
```
I'm fine getting rid of this convention. I think it was done in order to avoid flake8 errors, but we are now ignoring this whole file in the flake8 config, so no use in keeping it.

jmargeta replied:
Ah, ok, that's why! Thank you @sloria for the explanation, for doing the remaining changes, and for merging them in :)

sloria reviewed (changes requested):
Thanks @jmargeta for the PR!

Just had a few nits. I can take care of those myself.

sloria committed:
Clean up imports in compat.py

* Remove unnecessary try-except logic
* Remove unnecessary assignments
